cfme/cloud/stack.py

```diff
@@ -49,11 +48,14 @@ def edit_tags(self, tag, value):
                               'select_value': value},
              action=form_buttons.save)
         flash.assert_success_message('Tag edits were successfully saved')
+        self.get_tags(tag, value)
+
+    def get_tags(self, tag, value):
         sel.force_navigate('clouds_stack', context={'stack': self})
         row = sel.elements("//*[(self::th or self::td) and normalize-space(.)={}]/../.."
                      "//td[img[contains(@src, 'smarttag')]]".format(quoteattr("My Company Tags")))
-        tag = sel.text(row).strip()
-        assert tag == "Cost Center: Cost Center 001"
+        company_tag = sel.text(row).strip()
+        assert company_tag == tag.replace(" *", "") + ": " + value
```

Comment:
Agreed.

Also, when doing the check, it should be a proper exception, assertions are ignored under some conditions. Also, this is a good candidate for string interpolation with `.format`:

```python
if company_tag != "{}: {}".format(tag.replace(" *", ""), value):
    raise something other than AssertionError here
```